scimma_admin/hopskotch_auth/views.py

```diff
@@ -407,33 +406,32 @@ def manage_topic(request, topicname) -> HttpResponse:
     access_result = engine.get_groups_with_access_to_topic(request.user, topic)
     if not access_result:
         return redirect_with_error(request, "manage_topic", access_result.err(), 'index')
-    groups_added = access_result.ok()
+    existing_perms = access_result.ok()
 
     groups_result = engine.get_all_groups()
     if not groups_result:
         return redirect_with_error(request, "manage_topic", groups_result.err(), 'index')
     groups_available = groups_result.ok()
 
-    cleaned_added = []
-    # TODO: Convert cleaned_added to a dictionary. The key is the group name and the value is a dictionary with keys
-    #       Read and Write where those values are booleans of whether the read or write permission exists
-    cleaned_available = []
+    groups_with_access = {}
+    for perm in existing_perms:
+        if perm.principal not in groups_with_access:
+            groups_with_access[perm.principal] = {
+                "name": perm.principal.name,
+                "permissions": [],
+            }
+        groups_with_access[perm.principal]["permissions"].append(perm.operation._name_)
+    
+    other_groups = {}
     for group in groups_available:
-        is_added = False
-        for added in groups_added:
-            if group == added.principal:
-                is_added = True
-                break
-        if is_added:
-            perms_result = engine.get_group_permissions_for_topic(request.user, group.name, topicname)
-            if not perms_result:
-                return redirect_with_error(request, 'manage_topic', perms_result.err(), 'index')
-            ops = [op.operation for op in perms_result.ok()]
-            cleaned_added.append({'name': group.name,
-                                  'read': KafkaOperation.Read in ops,
-                                  'write': KafkaOperation.Write in ops})
-        else:
-            cleaned_available.append(group.name)
+        for op in [KafkaOperation.Read, KafkaOperation.Write]:
```

Comment:
here's another place your using operation/permission interchangeably

Reply:
It's really too late to change the name of the KafkaOperation type unless it's truly important, and while consistency is nice inside the code, I think it's less important that keeping the database schema stable.